File: src/main/java/PongDaemon.java
package le.yord.samples.java;

import com.typesafe.config.ConfigFactory;
import akka.kernel.Bootable;

import akka.actor.ActorSystem;
import akka.actor.Actor;
import akka.actor.Props;
import akka.actor.ActorRef;

public class PongDaemon implements Bootable {
  
  final ActorSystem system = ActorSystem.create("PongDaemon", ConfigFactory.load().getConfig("pongDaemon"));
  
  public static void main(String[] args) {
    new PongDaemon().startup();
  }
  
  public void startup() {
    system.actorOf(new Props(PongActor.class), "ponger");
    System.out.println("Started Pong Daemon!");
  }
  
  public void shutdown() {
    system.shutdown();
  }
  
}

File: src/main/scala/PongDaemon.scala
package le.yord.samples.scala

import com.typesafe.config.ConfigFactory
import akka.kernel.Bootable

import akka.actor.{ ActorSystem, Actor, Props, ActorRef }

object PongDaemon extends Bootable {
  
  val system = ActorSystem("PongDaemon", ConfigFactory.load.getConfig("pongDaemon"))
  
  def main(args: Array[String]): Unit = {
    startup()
  }
  
  def startup(): Unit = {
    val ponger: ActorRef = system.actorOf(Props[PongActor], "ponger")
    println("Started Pong Daemon!")
  }
  
  def shutdown(): Unit = {
    system.shutdown()
  }
  
}

class PongActor extends Actor {
  
  def receive = {
    case string: String => {
      println("PongActor: Received String(\""+string+"\") message!")
      val response = string.toUpperCase + " pong"
      sender ! response
      println("PongActor: Send String(\""+response+"\") message!")
    }
    case Stop => {
      println("PongActor: Received Stop message!")
      context.system.shutdown()
      println("Pong Daemon terminated!")
    }
  }
}